bluemira/codes/_freecadapi.py

```diff
@@ -1102,6 +1051,124 @@ def point_inside_shape(point, shape):
     return shape.isInside(vector, EPS, True)
 
 
+# ======================================================================================
+# Geometry checking tools
+# ======================================================================================
+
+
+def _edges_tangent(edge_1, edge_2):
+    """
+    Check if two adjacent edges are tangent to one another.
+    """
+    angle = edge_1.tangentAt(edge_1.LastParameter).getAngle(
+        edge_2.tangentAt(edge_2.FirstParameter)
+    )
+    return np.isclose(
+        angle,
+        0.0,
+        rtol=1e-4,
+        atol=1e-4,
+    )
+
+
+def _wire_edges_tangent(wire):
+    """
+    Check that all consecutive edges in a wire are tangent
+    """
+    if len(wire.Edges) <= 1:
+        return True
+
+    else:
+        edges_tangent = []
+        for i in range(len(wire.OrderedEdges) - 1):
+            edge_1 = wire.OrderedEdges[i]
+            edge_2 = wire.OrderedEdges[i + 1]
+            edges_tangent.append(_edges_tangent(edge_1, edge_2))
+
+    if wire.isClosed():
+        # Check last and first edge tangency
+        edges_tangent.append(_edges_tangent(wire.OrderedEdges[-1], wire.OrderedEdges[0]))
+
+    return all(edges_tangent)
+
+
+def _wire_is_planar(wire):
+    """
+    Check if a wire is planar.
+    """
+    try:
+        face = Part.Face(wire)
+    except Part.OCCError:
+        return False
+    return isinstance(face.Surface, Part.Plane)
+
+
+def _wire_is_straight(wire):
+    """
+    Check if a wire is a straight line.
+    """
+    if len(wire.Edges) == 1:
+        edge = wire.Edges[0]
+        if len(edge.Vertexes) == 2:
+            straight = dist_to_shape(edge.Vertexes[0], edge.Vertexes[1])[0]
+            if np.isclose(straight, wire.Length, rtol=EPS, atol=1e-8):
+                return True
+    return False
+
+
+def _is_wire_or_face(shape_type):
+    return shape_type == apiWire or shape_type == apiFace
+
+
+def _check_shapes_same_type(shapes):
+    """
+    Check that all the shapes are of the same type.
+    """
+    _type = type(shapes[0])
+    if not all(isinstance(s, _type) for s in shapes):
+        raise ValueError(f"All instances in {shapes} must be of the same type.")
+
+
+def _check_shapes_coplanar(shapes):
+    if not _shapes_are_coplanar(shapes):
+        raise ValueError(
+            f"Shapes of type {type(shapes[0])} are not co-planar; this operation does not support non-co-planar wires or faces."
```

Comment:
The error message considers that the shapes are all of the same type (of shapes[0]), but this is not a "requirement". Probably it should be changed.